### PR TITLE
[Doc] Add custom routes doc

### PR DESCRIPTION
**Pull request by fzaninotto**

### CODE REVIEW

File: docs/CustomRoutes.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+// in src/App.js
+import * as React from "react";
+import { Admin, CustomRoutes } from 'react-admin';
+import Foo from './foo';
+import Register from './register';
+
+const App = () => (
+    <Admin dataProvider={simpleRestProvider('http://path.to.my.api')}>
+        <CustomRoutes noLayout>
+            <RouteWithoutLayout path="/register" element={<Register />} />
```
Comment (WiXSL) [inline docs/CustomRoutes.md:69]:
```suggestion
            <Route path="/register" element={<Register />} />
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/CustomRoutes.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        </CustomRoutes>
+    </Admin>
+);
+
+export default App;
+```
+
+When a user browses to `/register`, the `<Register>` component will appear outside the defined Layout, leaving you the freedom to design the screen the way you want.
+
+**Tip**: Custom routes can be [a `<Redirect>` route](https://reacttraining.com/react-router/web/api/Redirect), too.
```
Comment (WiXSL) [inline docs/CustomRoutes.md:82]:
```suggestion
**Tip**: Custom routes can be [a `<Navigate>` route](https://reactrouter.com/docs/en/v6/api#navigate), too.
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

**Review by WiXSL — APPROVED** [review]

File: docs/CustomRoutes.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        <CardContent>
+            ...
+        </CardContent>
+    </Card>
+);
+
+export default Foo;
+```
+
+**Tip**: It's up to you to create a [custom menu](#menu) entry, or custom buttons, that lead to your custom pages.
```
[the remark this answers is not in the document]
Comment (fzaninotto) [reply docs/CustomRoutes.md:55]: I'll link to the Theming doc instead

File: docs/CustomRoutes.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+import Foo from './foo';
+import Bar from './bar';
+
+const App = () => (
+    <Admin dataProvider={simpleRestProvider('http://path.to.my.api')}>
+        <Resource name="posts" {...posts} />
+        <Resource name="comments" {...comments} />
+        <CustomRoutes>
+            <Route path="/foo" element={<Foo />} />
+            <Route path="/bar" element={<Bar />} />
```
Comment (djhi) [inline docs/CustomRoutes.md:25]: I suggest we stop using foo/bar in our examples. First, it's very far from real usages. Second the story behind those is horrible
```suggestion
            <Route path="/settings" element={<Settings />} />
            <Route path="/profile" element={<Profile />} />
```

File: docs/CustomRoutes.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+To register your own routes, pass one or several `<CustomRoutes>` elements as children of `<Admin>`. Declare as many [react-router-dom](https://reactrouter.com/docs/en/v6/api#routes-and-route) `<Route>` as you want inside them:
+
+```jsx
+// in src/App.js
+import * as React from "react";
+import { Admin, Resource, CustomRoutes } from 'react-admin';
+import posts from './posts';
+import comments from './comments';
+import Foo from './foo';
+import Bar from './bar';
```
Comment (djhi) [inline docs/CustomRoutes.md:17]:
```suggestion
import Settings from './Settings';
import Profile from './Profile';
```

File: docs/CustomRoutes.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            <Route path="/foo" element={<Foo />} />
+            <Route path="/bar" element={<Bar />} />
+        </CustomRoutes>
+    </Admin>
+);
+
+export default App;
+```
+
+Now, when a user browses to `/foo` or `/bar`, the components you defined will appear in the main part of the screen.
```
Comment (djhi) [inline docs/CustomRoutes.md:33]:
```suggestion
Now, when a user browses to `/settings` or `/profile`, the components you defined will appear in the main part of the screen.
```

File: docs/CustomRoutes.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+**Tip**: To look like other react-admin pages, your custom pages should have the following structure:
+
+```jsx
+// in src/Foo.js
```
Comment (djhi) [inline docs/CustomRoutes.md:38]:
```suggestion
// in src/Settings.js
```

File: docs/CustomRoutes.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+import * as React from "react";
+import { Card, CardContent } from '@mui/material';
+import { Title } from 'react-admin';
+
+const Foo = () => (
```
Comment (djhi) [inline docs/CustomRoutes.md:43]:
```suggestion
const Settings = () => (
```

File: docs/CustomRoutes.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    <Card>
+        <Title title="My Page" />
+        <CardContent>
+            ...
+        </CardContent>
+    </Card>
+);
+
+export default Foo;
```
Comment (djhi) [inline docs/CustomRoutes.md:52]:
```suggestion
export default Settings;
```

File: docs/CustomRoutes.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+**Tip**: It's up to you to create a [custom menu](./Theming.md#using-a-custom-menu) entry, or custom buttons, that lead to your custom pages.
+
+If you want a custom route to render without the layout (without the menu and the appBar), e.g. for registration screens, then provide the `noLayout` prop on the `<CustomRoutes>` element:
+
+```jsx
+// in src/App.js
+import * as React from "react";
+import { Admin, CustomRoutes } from 'react-admin';
+import Foo from './foo';
```
Comment (djhi) [inline docs/CustomRoutes.md:63]:
```suggestion
import Settings from './Settings';
```

File: docs/CustomRoutes.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+import Register from './register';
+
+const App = () => (
+    <Admin dataProvider={simpleRestProvider('http://path.to.my.api')}>
+        <CustomRoutes noLayout>
+            <Route path="/register" element={<Register />} />
+        </CustomRoutes>
+        <CustomRoutes noLayout>
+            <Route path="/foo" element={<Foo />} />
```
Comment (djhi) [inline docs/CustomRoutes.md:72]:
```suggestion
            <Route path="/settings" element={<Settings />} />
```

File: docs/Routing.md [hunk 12]
```diff
@@ -56,6 +56,33 @@ const Dashboard = () => {
...
+In addition to CRUD pages for resources, you can create as many routes as you want for your custom pages. Use [the `<CustomRoutes>` component](./CustomRoutes.md) to do so.
+
+```jsx
+// in src/App.js
+import * as React from "react";
+import { Admin, Resource, CustomRoutes } from 'react-admin';
+import posts from './posts';
+import comments from './comments';
+import Foo from './foo';
+import Bar from './bar';
```
Comment (djhi) [inline docs/Routing.md:70]:
```suggestion
import Settings from './Settings';
import Profile from './Profile';
```

File: docs/Routing.md [hunk 13]
```diff
@@ -56,6 +56,33 @@ const Dashboard = () => {
...
+
+const App = () => (
+    <Admin dataProvider={simpleRestProvider('http://path.to.my.api')}>
+        <Resource name="posts" {...posts} />
+        <Resource name="comments" {...comments} />
+        <CustomRoutes>
+            <Route path="/foo" element={<Foo />} />
+            <Route path="/bar" element={<Bar />} />
```
Comment (djhi) [inline docs/Routing.md:78]:
```suggestion
            <Route path="/settings" element={<Settings />} />
            <Route path="/profile" element={<Profile />} />
```

**Review by djhi — CHANGES_REQUESTED** [review]

**Pull request merged**